Schronisko_app.py

```diff
@@ -0,0 +1,129 @@
+from flask import Flask, request, render_template, g, flash, redirect, url_for, abort
+
+import sqlite3
+
+DATABASE = '/path/to/database.db'
+app = Flask(__name__)
+app.secret_key = "TjWnZq4t7w!z%C*F"
+app_info = {
+    "db_file": "database/schronisko.db"
+}
+"""
+create table animals(id integer primary key autoincrement, name varchar(35), category varchar (40), race varchar(50), 
+description text, admisson_date date not null default(date()));
+"""
+
+
+def get_db():
+    if not hasattr(g, 'sqlite_db'):
+        conn = sqlite3.connect(app_info['db_file'])
+        conn.row_factory = sqlite3.Row
+        g.sqlite_db = conn
+    return g.sqlite_db
+
+
+@app.teardown_appcontext
+def close_db(error):
+    if hasattr(g, 'sqlite_db'):
+        g.sqlite_db.close()
+
+
+@app.route("/", methods=["GET"])
+def home():
+    return render_template("home.html")
+
+
+@app.route("/add", methods=['GET', 'POST'])
+def add_animal():
+    if request.method == 'POST':
+        name = request.form['name']
+        category = request.form['category']
+        race = request.form['race']
+        description = request.form['description']
+
+        if not name:
+            flash('Imię jest wymagane')
+        elif not category:
+            flash('Kategoria jest wymagana')
+        elif not race:
+            flash('Rasa jest wymagana')
```

Comment:
We can practice a more separated MVC solution where REST API doesn't render or flash things, it only returns serializable responses. Please check this out: https://www.youtube.com/watch?v=GMppyAPbLYk . In such a solution you'd return a reponse with error status and fronted would handle it